### PR TITLE
Fix #1466 Account awaiting activation notification settings

### LINKED ISSUE #1007: account awaiting activation bug
i install mybb 1.8 beta 3
i register a user but got an mysql error:
see picture:

now with ADMIN account i have error on top:
There is 1 account awaiting activation. Please go to your ACP to activate the users.

and in Admin_CP:
2 Registered Users
2 Active Users
2 Registrations Today
1 Awaiting Activation

when i click on:
1 Awaiting Activation
i got this error:
No users were found matching the specified search criteria. Please modify your search criteria and try again. 

------

and
Registration Method: Send Email Verification
also i use Activation url from user email.
and user already marked as registered

------

i think this is a bug. but if i have free time, test it with more registrations.

Original thread: [account awaiting activation bug](http://community.mybb.com/thread-156406.html)

### LINKED ISSUE #1466: Account awaiting activation notification settings
The notification bar should have some settings or be affected by current ones.

- [x] Turn on/off setting.
- [x] ACP link?

__Reference:__ http://community.mybb.com/thread-158427.html

### PR DESCRIPTION
**Pull request by Sama34**
Fix #1466 Account awaiting activation notification settings
Fixes #1007 account awaiting activation bug

IIRC simply updating the `settings.php` file will update the cache and everything.

### CODE REVIEW

**Comment by PirataNervo** [comment]
The code seems fine to be but needs to be tested before merging. At first glance however, it seems good :)

**Comment by Destroy666x** [comment]
I don't think `$l['awaiting_message_link']` will have a good result since both `sprintf()` and concatenation are used at the same time:
https://github.com/mybb/mybb/pull/1710/files#diff-89024b9d05bdf5fd4144014492cc7cafR763

Also, the `sprintf()` is weird, ` .` is added after the 1st sentence for no reason: https://github.com/mybb/mybb/pull/1710/files#diff-3ea9773affe02a0fa78965ca14e068fcR541

**Comment by Sama34** [comment]
I see what you mean.

**Comment by Sama34** [comment]
That should fix it.

**Comment by Sama34** [comment]
Unless somebody confirms it works I will be merging this myself after re-testing it my self.

I can't work on new issues unless I update my repository, and I know my PRs get f*cked after that..

File: global.php [hunk 1]
```diff
@@ -719,10 +719,16 @@
 	eval('$pm_notice = "'.$templates->get('global_pm_alert').'";');
 }
 
-if($mybb->usergroup['cancp'] == 1)
+if($mybb->settings['awactialert'] == 1 && $mybb->usergroup['cancp'] == 1)
 {
 	$awaitingusers = $cache->read('awaitingactivation');
 
+	if(isset($awaitingusers['time']) && (int)$awaitingusers['time'] < TIME_NOW + 86400)
```
Comment (Destroy666x) [inline global.php]: `(int)` not needed here, `TIME_NOW` is always an integer.

**Comment by Sama34** [comment]
There it is.

**Comment by Destroy666x** [comment]
Except that the cache update could be added to few more places, it looks ok.

**Pull request merged**